base/applications/rapps/include/rosui.h

```diff
@@ -21,31 +25,34 @@ class CPointerArray
         m_hDpa = DPA_Create(GrowthRate);
     }
 
-    ~CPointerArray()
+    virtual ~CPointerArray()
     {
+        // Here, s_OnRemoveItem() can only call our own OnRemoveItem() implementation...
+        // unless we use the CRTP construct...
         DPA_DestroyCallback(m_hDpa, s_OnRemoveItem, this);
     }
 
 private:
     static INT CALLBACK s_OnRemoveItem(PVOID ptr, PVOID context)
     {
-        CPointerArray * self = (CPointerArray*) context;
-        return (INT) self->OnRemoveItem(reinterpret_cast<T*>(ptr));
+        // TDeriv* pThis = static_cast<TDeriv*>(context);
+        CPointerArray* pThis = (CPointerArray*)context;
```

Comment:
```suggestion
        auto pThis = reinterpret_cast<CPointerArray*>(context);
```

Reply:
I won't use `auto` as I want this code to be compilable with e.g. MSVC 2010.